packages/firebase_admob/example/android/app/src/main/java/io/flutter/plugins/firebaseadmobexample/EmbeddingV1Activity.java

```diff
@@ -7,11 +7,15 @@
 import android.os.Bundle;
 import io.flutter.app.FlutterActivity;
 import io.flutter.plugins.GeneratedPluginRegistrant;
+import io.flutter.plugins.firebaseadmob.FirebaseAdMobPlugin;
+import io.flutter.plugins.firebaseadmob.FirebaseAdMobPlugin.NativeAdFactory;
 
 public class EmbeddingV1Activity extends FlutterActivity {
   @Override
   protected void onCreate(Bundle savedInstanceState) {
     super.onCreate(savedInstanceState);
     GeneratedPluginRegistrant.registerWith(this);
+    final NativeAdFactory factory = new NativeAdFactoryExample(getLayoutInflater());
+    FirebaseAdMobPlugin.registerNativeAdFactory(this, "adFactoryExample", factory);
```

Comment:
We should probably call unregisterNativeAdFactory in `onStop`?

Reply:
According to [android documentation](https://developer.android.com/guide/components/activities/activity-lifecycle#onstop) for `onStop`, it is called when the app is no longer visible. For example, a would answer a call and then return to the app. This would call `onStop`, then `onRestart`.  NativeAds would no longer work, but the app would be visible. 

It also states: "You should also use onStop() to perform relatively CPU-intensive shutdown operations." I don't believe the NativeAdFactory qualifies for this.